anaDACScan.py

```diff
@@ -0,0 +1,236 @@
+             
+    # Loop over events in the tree and fill plots
+    for event in dacScanFile.dacScanTree:
+        oh = event.link
+        vfat = event.vfatN
+
+        #the reversal of x and y is intended - we want to plot the nameX variable on the y-axis and the nameY variable on the x-axis
+        calibrated_DAC_value=calInfo[oh]['slope'][vfat]*event.dacValX+calInfo[oh]['intercept'][vfat]
+        calibrated_DAC_error=calInfo[oh]['slope'][vfat]*event.dacValX_Err+calInfo[oh]['intercept'][vfat]
+
+        if args.assignXErrors:
+            dict_dacScanResults[oh][vfat].SetPoint(dict_dacScanResults[oh][vfat].GetN(),event.dacValY,calibrated_DAC_value)
+            dict_dacScanResults[oh][vfat].SetPointError(dict_dacScanResults[oh][vfat].GetN()-1,event.dacValY_Err,calibrated_DAC_error)
+        else:    
+            dict_dacScanResults[oh][vfat].SetPoint(dict_dacScanResults[oh][vfat].GetN(),event.dacValY,calibrated_DAC_value)
+            dict_dacScanResults[oh][vfat].SetPointError(dict_dacScanResults[oh][vfat].GetN()-1,event.dacValY_Err,0)
+
+    # Fit the TGraphErrors objects    
+    for oh in ohArray:
+        for vfat in range(0,24):
+            dict_dacScanFuncs[oh][vfat] = r.TF1("DAC Scan Function","[0]*x+[1]")
+            dict_dacScanResults[oh][vfat].Fit(dict_dacScanFuncs[oh][vfat]) 
+
+    nominal = -1        
+
+    #from Tables 9 and 10 of the VFAT3 manual 
+    if nameX == "CFG_THR_ARM_DAC":
+        nominal = 3.2
+    else:
+        print("Error: unexpected value of nameX: '%s'"%nameX)
+        exit(1)
+        
+    # Determine DAC values to achieve recommended bias voltage and current settings
+    graph_dacVals = {}
+    tree_dacVals = {}
+    dict_dacVals = nesteddict()
+    for oh in ohArray:
+        graph_dacVals[oh] = r.TGraph()
+        tree_dacVals[oh] = r.TTree()
+        vfatN_branch = np.zeros(1, dtype=int)
+        dacVal_branch = np.zeros(1, dtype=int)
+        tree_dacVals[oh].Branch('vfatN',vfatN_branch,'vfatN/i')
+        tree_dacVals[oh].Branch('dacVal',dacVal_branch,'dacVal/i')
+        for vfat in range(0,24):
+             dict_dacVals[oh][vfat] = dict_dacScanResults[oh][vfat].Eval(nominal)
+             graph_dacVals[oh].SetPoint(graph_dacVals[oh].GetN(),vfat,dict_dacVals[oh][vfat])
+             vfatN_branch[0] = vfat
+             dacVal_branch[0] = dict_dacVals[oh][vfat]
+             tree_dacVals[oh].Fill() 
+             
+    # Write out the dacVal results to both a file and the terminal         
+    print( "| OH | vfatN | dacVal |")
+    print( "| -- | ----- | ------ |")        
+    
+    for oh in ohArray:
+        outputFiles[oh].cd()
+        graph_dacVals[oh].Write("dacValsOH"+str(oh))
+        tree_dacVals[oh].Write("dacValsOH"+str(oh))
+        for vfat in range(0,24):
+            print("| {0} | {1}  | {2} | ".format(
+                oh,
+                vfat,
+                dict_dacVals[oh][vfat]
+            ))
```

Comment:
Typically `TFile`s produced by scripts in `gem-plotting-tools` are organized with subdirectories to be easier to find objects within them.  The typical subdirectory structure is:

```
VFAT0
VFAT1
...
VFAT22
VFAT23
Summary
```

Where Summary contains summary information (e.g. lines 231-236).

Please add this structure to the output `TFile` objects.

Reply:
Why would you store this in the output root file?  You have the TF1 in the output root file, that's all that's needed.  Moreover the DAC value is an integer, not a float.

Reply:
When I evaluate the TF1 at the nominal current or voltage, I get a float. That is the value I was planning to store in the output root file. It is written to the output text file and the terminal, but not stored in the root file. Furthermore, how should it be converted to an int? Just use floor?

Reply:
But what is the purpose of storing it in the output `TFile`?  You have the `TF1` already which is much more useful.  Don't store it in the output `TFile`, you should be printing this information to a separate text file.

And as for converting just use `int()` which will handle the conversion and round appropriately.

Reply:
done